lms/djangoapps/courseware/views.py

```diff
@@ -656,7 +666,13 @@ def jump_to(_request, course_id, location):
     except InvalidKeyError:
         raise Http404(u"Invalid course_key or usage_key")
     try:
+        user = _request.user
         redirect_url = get_redirect_url(course_key, usage_key)
+        if GlobalStaff().has_user(user) and not CourseEnrollment.is_enrolled(user, course_key):
```

Comment:
eventually, this is going to look exactly like the code block in `_index_bulk_op`. Let's abstract it out :)